platform/sensor-stream-manager/sensor-stream-manager.js

```diff
@@ -1,72 +1,208 @@
 const fs = require("fs-extra");
 const mqtt = require("mqtt");
 const utils = require("../../utils");
+const fetch = require('node-fetch');
 
 // TODO: add mqtt-data-collector logic to SSM.
 const MessagingService = require('../messaging-service');
 
-console.log("[INFO] Initialize app-manager...");
+/**
+ * This function gets the gateway's ip.
+ * @returns {string} - gateway's ip
+ */
+function getGatewayIp() {
+    let ip = utils.getIPAddress();
+    if(!ip) {
+        console.error("[ERROR] No IP address found. Please ensure the config files are set properly.");
+        process.exit(1);
+    }
+    console.log(`[INFO] Gateway's ip address is ${ip}`);
+    return ip;
+}
```

Comment:
This function is not simple enough. If the function is to get the Gateway's IP, it should get an IP and it should not exit the program if it doesn't get it. Exiting should not be the function's duty. That should be done in the main program. 
So if you remove the exit part, then this function would be reduced to just this line:
let ip = utils.getIPAddress();
So then it's better to remove the whole function. and just do this directly from the main program itself.